### ISSUE TITLE
Pool::run should return impl Future that is Send

### ISSUE THREAD

**Issue by vorot93** [khuey/bb8#5]
Boxing is no longer necessary. On the other hand the returned future being `Send` will give much more freedom to DB result consumers.

**Comment by khuey** [comment]
Adding `Send` to the future returned from `Pool::run` would require some internal changes, and I don't really see why it's useful.

Why don't you just do something like
```rust
  let (tx, rx) = oneshot::channel();
  let f = pool.run(my_closure)
    .then(|r| tx.send(r).unwrap());
  current_thread_executor.dispatch(f);
  let f = rx.and_then(blah_blah_blah);
  do_something_requiring_send(f);
```
?

**Comment by vorot93** [comment]
This may be possible but adds boilerplate to the detriment of user ergonomics.

**Comment by khuey** [comment]
Sure, but there's also a cost to providing `Send` for those who don't need it. What are you doing that requires `Send`?

**Comment by Daniel-B-Smith** [comment]
Just seeing this now (after creating #10). All futures that generate Hyper responses require `Send`, so the boilerplate from your comment would be needed for any code that generated Hyper responses.

**Comment by khuey** [comment]
This is partially fixed by #10, and the rest will be fixed if #8 is merged, so I'm going to close this.
